### PR TITLE
29 datadescription class

### LINKED ISSUE #29: DataDescription class
Class to load the DataDescription json, potentially processing it a little. Exposes some methods/properties to make handling the data a bit easier.

Attributes
- `description`

Properties
- `num_features`
- `num_encoded_features`

Methods
- `num_encoded` (instead of as a property, to allow different arguments, e.g. for ordinals to be treated as categorical or not)
- `__iter__` to iterate over the columns, returning each `cdict`.

### PR DESCRIPTION
**Pull request by JamesJordon**
`DataDescription` class implemented according to #29 . This closes #29 .

### CODE REVIEW

File: prive/datasets/data_description.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+
+
+class DataDescription:
+    def __init__(self, description):
+        self.schema = description
```
Comment (triangle-man) [inline prive/datasets/data_description.py:5]: Maybe should be
```
def __init__(self, schema):
    self.schema = schema
```
?

File: prive/datasets/data_description.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        """
+        int : Number of dimensions the data would have if encoded. This assumes
+        ordered and infinite variables will have one dimension, and only finite,
+        unordered variables would be one-hot encoded, where they will require
+        one dimension per category.
+
+        """
+        nfeatures = 0
+
+        for cdict in self:
```
Comment (triangle-man) [inline prive/datasets/data_description.py:26]: Slightly confused by python here. Why isn't this `for cdict in self.schema`? 

Also ... `schema` will be a list, not a dict (if indeed that's what was meant here, but as noted I am confused.)

File: prive/datasets/data_description.py (hunk 1 above)
Comment (JamesJordon) [reply prive/datasets/data_description.py:5]: Agreed

File: prive/datasets/data_description.py (hunk 2 above)
Comment (triangle-man) [reply prive/datasets/data_description.py:26]: Okay, going to leave this in for now, make the change above, and merge.

**Review by triangle-man — APPROVED** [review]

File: prive/datasets/data_description.py (hunk 2 above)
Comment (JamesJordon) [reply prive/datasets/data_description.py:26]: So the implementation of `__getitem__` below essentially allows us to iterate over `self.schema` by just calling iteration over `self`. This class is essentially a list with some extra methods and some alternative indexing (can index by column name). This saves us from having to always write `description.schema`.

**Pull request merged**